infrastructure/modules/aws/eks/instance_refresh.tf

```diff
@@ -137,9 +148,9 @@ resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_spot" {
 # ensures that node termination does not require the lifecycle action to be completed,
 # and thus allows the ASG to be destroyed cleanly.
 resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler" {
-  count                  = length(module.eks.workers_asg_names)
+  count                  = length(local.eks_worker_group)
```

Comment:
```suggestion
  for_each                  = toset(module.eks.self_managed_node_groups_autoscaling_group_names)
```

Reply:
ok nice

Reply:
hum can't : 
│ Error: Invalid for_each argument
│
│   on ../../../modules/aws/eks/instance_refresh.tf line 151, in resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler":
│  151:   for_each               = toset(module.eks.self_managed_node_groups_autoscaling_group_names)
│     ├────────────────
│     │ module.eks.self_managed_node_groups_autoscaling_group_names is a list of string, known only after apply